components/abstractions/src/main/java/com/microsoft/kiota/Headers.java

```diff
@@ -0,0 +1,53 @@
+package com.microsoft.kiota;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+
+class Headers extends CaseInsensitiveMap {
```

Comment:
we should either:
- have a copy method
- in API exception, have getters and setters and check that nobody is assigning null
Otherwise there's no way we can enforce the nonnull attribute without the final modifier (my last commit)